File: tpl/src/rngsse2/examples/examples_CPU_FORTRAN/lfsr113.f90

PROGRAM LFSR113_Example

   PARAMETER (N=100000000)
   INTEGER i,LFSR113_generate,sum
   REAL val,rsum

   TYPE LFSR113_state
     INTEGER z(4)
   END TYPE LFSR113_state
   TYPE(LFSR113_state) state

   CALL LFSR113_init ( state )
   CALL LFSR113_Print_State (state)

   sum = 0

   DO i=1,N
     sum = sum + LFSR113_generate (state)
   END DO

   WRITE(*,1) N
   rsum = sum/4294967296.
   val = LFSR113_generate (state)/4294967296.
   IF (rsum<0) THEN
     rsum=rsum+1
   ENDIF
   IF (val<0) THEN
     val=val+1
   ENDIF
   WRITE(*,2) rsum
   WRITE(*,3) val

1  FORMAT(I9," LFSR113 pseudorandom numbers generated.")
2  FORMAT("Fractional part of the total sum of generated numbers: ",F8.6)
3  FORMAT("Next output value: ",F8.6)

END PROGRAM LFSR113_Example

File: examples/examples_CPU_C/lfsr113.c

#include<stdio.h>
#include<lfsr113.h>

#define NN       100000000UL

int main(void){ 
   long i; unsigned int sum=0;
   lfsr113_state state;
   lfsr113_init_(&state);
   lfsr113_print_state_(&state);
   for(i=0;i<NN;i++) sum+=lfsr113_generate_(&state);
   printf("%ld LFSR113 pseudorandom numbers generated.\n",NN);
   printf("Fractional part of the total sum of generated numbers: %f\n",sum/4294967296.);
   printf("Next output value: %f\n",lfsr113_generate_(&state)/4294967296.);
   return 0;
}
